### PR TITLE
Feature/config page

### PR DESCRIPTION
**Pull request by sumukhmarathe**
Made configuration of connection string and device id dynamic. Created a simple form for user input. Refer to updated Readme for new steps.

### CODE REVIEW

**Comment by ryanwinter** [comment]
Cool, let me try this out!

File: app/HTTPServer.js [hunk 1]
```diff
@@ -1,42 +1,107 @@
...
-        if (query.type === TYPE_PROPERTY)
+        const { HostName, SharedAccessKeyName, SharedAccessKey } = parseConnectionString(
+            req.body.connectionstring
+        )
+        // Verify that the required info is in the connection string.
+        if (!HostName || !SharedAccessKey || !SharedAccessKeyName)
         {
-            UpdateDigitalTwin(dtServiceclient, query.componentName, query.methodName, query.value)
-            console.log('Updating Property..')
+            body = 'Invalid IoT Hub Connection String! <a href="http://localhost:8080/configuration-form">Reopen Solution Sample Configuration</a>'
```
Comment (ryanwinter) [inline app/HTTPServer.js:42]: Update this to http://localhost:8080 if the above change request makes sense.

**Review by ryanwinter — CHANGES_REQUESTED** [review]
I think adding a dashboard link from grafana to the config page would be helpful here, then if people accidently miss the docs they will at least hopefully see a configure link on the dashboard.

https://grafana.com/docs/grafana/latest/linking/dashboard-links/

File: README.md [hunk 2]
```diff
@@ -92,10 +66,58 @@ NOTE: If you do not want to use VS code and the docker extension, you can run th
...
+  ![Configure Azure IoT Hub Connection String](./media/config_landing_page_conn_string.png)
+
+* Enter your device ID
+    * Example -  `MyMXChipDevice`
+
+  ![Configure Device ID](./media/config_landing_page_device_id.png)
+
+* Click on Save Configuration
+
+  ![Save Config](./media/config_landing_page_save.png)
```
[the remark this answers is not in the document]
Comment (sumukhmarathe) [reply README.md:92]: have 2 screenshots now. Let me know if you think first one is necessary

File: README.md [hunk 3]
```diff
@@ -92,10 +66,58 @@ NOTE: If you do not want to use VS code and the docker extension, you can run th
...
+
```
[the remark this answers is not in the document]
Comment (sumukhmarathe) [reply README.md:94]: Changed cursor type to "pointer" from "grabbing"

File: README.md [hunk 4]
```diff
@@ -92,10 +66,58 @@ NOTE: If you do not want to use VS code and the docker extension, you can run th
...
+  >Click on the link to navigate to the solution sample landing page
+
+  ![Successful Configuration](./media/config_landing_page_success_redirect.png)
+
+
+* If an illegal IoT Hub Connection String is provided, you might see the following error message
+
+  >Re-open and re-fill the configuration settings form and ensure that a valid IoT Hub Connection String is provided
+
+  ![Failure in Configuration](./media/config_landing_page_error_redirect.png)
```
[the remark this answers is not in the document]
Comment (sumukhmarathe) [reply README.md:106]: done

File: app/HTTPServer.js [hunk 5]
```diff
@@ -1,42 +1,107 @@
...
-    http.createServer(function (req, res)
-    {
-        res.writeHead(200, {})
-        res.end()
+    const app = express()
+    app.use(express.urlencoded())
 
-        console.log('Received URL: %s', req.url)
-        const query = url.parse(req.url, true).query
+    app.get('/configuration-form', function (req, res)
```
[the remark this answers is not in the document]
Comment (sumukhmarathe) [reply app/HTTPServer.js:29]: Kept form at /configuration-form but added a redirect at root which will take user to /configuration-form

**Comment by sumukhmarathe** [comment]
Hey Ryan, can you review the changes done?

File: app/HTTPServer.js (hunk 1 above)
Comment (ryanwinter) [reply app/HTTPServer.js:42]: I think this caused a problem because the commands sent from the Grafana are sent to this top level URL.

Comment (ryanwinter) [reply app/HTTPServer.js:42]: I reverted this back because it was blocking my testing.

Comment (ryanwinter) [reply app/HTTPServer.js:42]: Here is where this URL is used: https://github.com/Azure/Verified-Telemetry-Solution-Sample/blob/9c62cb0f48249cfc8470a660319ff7e3e2518895/grafana/provisioning/dashboards/VerifiedTelemetryGSGDashboard.json#L206

**Comment by ryanwinter** [comment]
> Hey Ryan, can you review the changes done?

Done, sorry for slow response

**Review by ryanwinter — APPROVED** [review]

**Pull request merged**